dataprofiler/tests/data_readers/test_csv_data.py

```diff
@@ -473,14 +321,25 @@ def test_header_check_files(self):
 
         # add some more files to the list to test the header detection
         # these files have some first lines which are not the header
-        input_file_names = self.input_file_names
-        for input_file in input_file_names:
-            file_encoding = data_utils.detect_file_encoding(input_file['path'])
-            with open(input_file['path'], encoding=file_encoding) as csvfile:
+        for input_file in self.input_file_names:
+            with open(input_file['path'], encoding=input_file['encoding']) as csvfile:
                 data_as_str = ''.join(list(islice(csvfile, 5)))
             header_line = CSVData._guess_header_row(data_as_str, input_file['delimiter'])
             self.assertIn(header_line, input_file['has_header'], input_file['path'])
 
+        for input_buf in self.buffer_list:
+            # BytesIO is wrapped to use the isslice function to read it
+            if isinstance(input_buf['path'], BytesIO):
+                input_buf['path'] = TextIOWrapper(input_buf['path'], encoding=input_buf['encoding'])
```

Comment:
should mention that without being wrapped, the BytesIO object would get closed by the pandas reader in `py3.6`